nodeup/pkg/model/containerd.go

```diff
@@ -293,21 +309,21 @@ func (b *ContainerdBuilder) buildSystemdService() *nodetasks.Service {
 	manifest.Set("Service", "ExecStartPre", "-/sbin/modprobe overlay")
 	manifest.Set("Service", "ExecStart", "/usr/bin/containerd -c /etc/containerd/config-kops.toml \"$CONTAINERD_OPTS\"")
 
-	// kill only the containerd process, not all processes in the cgroup
-	manifest.Set("Service", "KillMode", "process")
+	manifest.Set("Service", "Restart", "always")
+	manifest.Set("Service", "RestartSec", "5")
+
 	// set delegate yes so that systemd does not reset the cgroups of containerd containers
 	manifest.Set("Service", "Delegate", "yes")
+	// kill only the containerd process, not all processes in the cgroup
+	manifest.Set("Service", "KillMode", "process")
+	// make killing of processes of this unit under memory pressure very unlikely
+	manifest.Set("Service", "OOMScoreAdjust", "-999")
 
 	manifest.Set("Service", "LimitNOFILE", "1048576")
 	manifest.Set("Service", "LimitNPROC", "infinity")
 	manifest.Set("Service", "LimitCORE", "infinity")
 	manifest.Set("Service", "TasksMax", "infinity")
 
-	manifest.Set("Service", "Restart", "always")
-	manifest.Set("Service", "RestartSec", "2s")
-	manifest.Set("Service", "StartLimitInterval", "0")
```

Comment:
We might actually need these to prevent systemd marking the unit as failed and not always restarting e.g. https://serverfault.com/questions/736624/systemd-service-automatic-restart-after-startlimitinterval

However, I think with RestartSec = 5 we're OK with the default values, but it is possible that systemd is not running with the default values.

Also annoying is that the field was renamed from StartLimitInterval to StartLimitIntervalSec at some stage; I don't know if any of our OSes uses StartLimitInterval.

We might need a helper function - so we can probably leave this as-is for now :-)

Reply:
I also saw that Kops doesn't use `StartLimitInterval` anywhere. Besides the name, it was also moved from `[Service]` to `[Unit]` at some point.
https://lists.freedesktop.org/archives/systemd-devel/2017-July/039255.html

Unless someone will complain, I also think we can skip it for now.